### PR TITLE
Fix API doc

### LINKED ISSUE #38: API documentation not working on RTD
The API documentation didn't actually populate.

https://pangeo-forge.readthedocs.io/en/latest/api.html

I suspect pangeo-forge is not being installed properly on RTD. Can we install it cheaply without all the dependencies?

### PR DESCRIPTION
**Pull request by davidbrochart**
Fixes #38

See https://pangeo-forge--46.org.readthedocs.build/en/46/api.html

There are other errors:
```
WARNING: autodoc: failed to import class 'recipe.BaseRecipe' from module 'pangeo_forge'; the following exception was raised:
cannot import name 'MultiStagePipeline' from 'rechunker.types' (/home/david/mambaforge/envs/pangeo-forge-dev/lib/python3.9/site-packages/rechunker/types.py) 
WARNING: autodoc: failed to import class 'recipe.NetCDFtoZarrSequentialRecipe' from module 'pangeo_forge'; the following exception was raised:
cannot import name 'MultiStagePipeline' from 'rechunker.types' (/home/david/mambaforge/envs/pangeo-forge-dev/lib/python3.9/site-packages/rechunker/types.py) 
WARNING: autodoc: failed to import class 'executors.PythonPipelineExecutor' from module 'pangeo_forge'; the following exception was raised:
cannot import name 'DaskPipelineExecutor' from 'rechunker.executors' (/home/david/mambaforge/envs/pangeo-forge-dev/lib/python3.9/site-packages/rechunker/executors/__init__.py)                                                                                                                                           
WARNING: autodoc: failed to import class 'executors.DaskPipelineExecutor' from module 'pangeo_forge'; the following exception was raised:
cannot import name 'DaskPipelineExecutor' from 'rechunker.executors' (/home/david/mambaforge/envs/pangeo-forge-dev/lib/python3.9/site-packages/rechunker/executors/__init__.py)                                                                                                                                           
WARNING: autodoc: failed to import class 'executors.PrefectPipelineExecutor' from module 'pangeo_forge'; the following exception was raised:
cannot import name 'DaskPipelineExecutor' from 'rechunker.executors' (/home/david/mambaforge/envs/pangeo-forge-dev/lib/python3.9/site-packages/rechunker/executors/__init__.py)
```
Indeed, it looks like `MultiStagePipeline` and `DaskPipelineExecutor` don't exist in rechunker.

### CODE REVIEW

**Comment by charlesbluca** [comment]
Is is possible to circumvent the `sys.path` trick by using `pip install -e` to install the package from the cloned repo and then import it in `conf.py`? This is seemingly what [xarray does for their RTD builds](https://github.com/pydata/xarray/blob/master/ci/requirements/doc.yml), although they specify their env with a YAML file instead of `requirements.txt` so not exactly sure how it would translate:

```
name: xarray-docs
channels:
    ...
dependencies:
    ...
  - pip
  - pip:
      - scanpydoc
      # relative to this file. Needs to be editable to be accepted.
      - -e ../..
```

**Comment by davidbrochart** [comment]
Let's try, thanks for hint!

**Comment by charlesbluca** [comment]
Glad to see it worked!

For the other errors, I assume that we need to change `PythonPipelineExecutor`, ... to `PythonExecutor`, ... to match up with the executors in rechunker, although I'm unsure about the `MultiStagePipeline` - @rabernat, is an addition you made to rechunker that isn't yet published?

**Comment by rabernat** [comment]
😁  Thanks so much @davidbrochart!

> is this an addition you made to rechunker that isn't yet published?

Bingo. We are pointing at this PR (https://github.com/pangeo-data/rechunker/pull/77), which is still unmerged.

I have this in the CI, so presumably we need to do something similar in the docs:
https://github.com/pangeo-forge/pangeo-forge/blob/f9b698a35a62ab169bce546b15fef9722c70cd90/ci/py3.8.yml#L31

Hope to merge that PR this week. Just needs a bit of real-world testing ferst.

File: docs/requirements.txt [hunk 1]
```diff
@@ -5,3 +5,6 @@ myst-parser
 myst-nb
 sphinx-copybutton
 sphinx-autodoc-typehints
+fsspec
+rechunker
```
Comment (charlesbluca) [inline docs/requirements.txt:9]:
```suggestion
git+https://github.com/rabernat/rechunker.git@refactor-executors
```

Why don't we go with the dev branch of rechunker for now to match CI, and once that is merged and published we can switch to the latest stable version.

Comment (rabernat) [reply docs/requirements.txt:9]:
```suggestion
git+https://github.com/pangeo-data/rechunker.git
```
The PR has been merged to master

Comment (charlesbluca) [inline docs/requirements.txt:9]:
```suggestion
git+https://github.com/pangeo-data/rechunker.git
```

Ryan's PR has been merged into the dev branch, so using that should fix the API build until the changes are incorporated into the next stable release.

Sorry, didn't mean to delete the original review!

**Comment by rabernat** [comment]
Executor docs still not working

```
WARNING: autodoc: failed to import class 'executors.PythonPipelineExecutor' from module 'pangeo_forge'; the following exception was raised:
No module named 'prefect'
WARNING: autodoc: failed to import class 'executors.DaskPipelineExecutor' from module 'pangeo_forge'; the following exception was raised:
No module named 'prefect'
WARNING: autodoc: failed to import class 'executors.PrefectPipelineExecutor' from module 'pangeo_forge'; the following exception was raised:
No module named 'prefect'
```

I guess we need prefect in the doc env.

**Comment by rabernat** [comment]
🎉

**Pull request merged**